### DISCUSSION TITLE
Why isn't the latest the default?

### DISCUSSION THREAD

**Discussion by rart** [yarnpkg/berry#5809] [General]
In the [getting started install instructions](https://yarnpkg.com/getting-started/install), it seems to assume you've already installed the `yarn` binary on your system and there's no reference to how to do so. The only from-scratch install instructions seem to be https://classic.yarnpkg.com/lang/en/docs/install (installs yarn _classic_).

When installing yarn (e.g. `npm install --global yarn`), at this time of writing, it installs `v1.22.19`. Initialising with `yarn init` — with no additional args — would initialise the project with the `1.x`. Even doing `yarn set version latest` would still keep the `1.x` version on the project.
```
$ yarn set version latest
Warning: Your current Yarn binary is currently Yarn 1.22.19; to avoid potential breaking changes, 'set version latest' won't receive upgrades past the 1.22.x branch.
         To upgrade to the latest versions, run yarn set version stable instead. Sorry for the inconvenience.
```

Why isn't the actual latest (currently 3.x) the default all across? 
Can the latest be installed globally if I'm not making use of _yarn classic_?
What's the right way of thinking about all this?

For context, I'm not at all new to yarn. I just find this... confusing.

Thanks 🙂

**Comment by clemyan** [comment]
> In the [getting started install instructions](https://yarnpkg.com/getting-started/install), it seems to assume you've already installed the yarn binary on your system and there's no reference to how to do so.

No, it only assumes you have a supported version of `node` installed. `corepack enable` *is* how you install the `yarn` binary

----

> When installing yarn (e.g. npm install --global yarn), at this time of writing, it installs v1.22.19

> Why isn't the actual latest (currently 3.x) the default all across?

To quote the installation guide you linked:

> The preferred way to manage Yarn is by-project and through [Corepack](https://nodejs.org/dist/latest/docs/api/corepack.html), a tool shipped by default with Node.js. Modern releases of Yarn aren't meant to be installed globally, or from npm.

See also https://yarnpkg.com/getting-started/qa#why-is-the-yarn-package-on-npm-still-on-1x

**Reply by rart** [reply]
I assumed it assumes `yarn` is pre installed because in the instructions it's invoking `yarn` without explaining where the binary came from. 

> 1. Start by enabling Corepack, if it isn't already:
corepack enable
>2. Then initialize a new project:
yarn init -2

Reading the [Corepack page](https://nodejs.org/dist/latest/docs/api/corepack.html) clarifies a bit:
> It exposes binary proxies for each [supported package manager](https://nodejs.org/dist/latest/docs/api/corepack.html#supported-package-managers) that, when called, will identify whatever package manager is configured for the current project, transparently install it if needed, and finally run it without requiring explicit user interactions.

So, if one has never installed `yarn` globally, simply doing the `corepack enable` would allow for running `yarn init -2`? Corepack makes the yarn binary available?

Is running `corepack enable` something that works globally, or its enabled in the directory where its ran?

If one has done a global npm install of yarn, would the recommendation be to `npm uninstall --global yarn` and continue to use it via Corepack?

Thanks!

**Reply by clemyan** [reply]
> So, if one has never installed yarn globally, simply doing the corepack enable would allow for running yarn init -2? Corepack makes the yarn binary available?

Yes

> Is running corepack enable something that works globally, or its enabled in the directory where its ran?

Enabling corepack works globally. It puts a `yarn` binary next to the `node` binary which you probably have in your `PATH`. As long as you are invoking that binary the corepack mechanism takes effect.

Configuring a project to use a package manager (e.g. via `corepack use` or manually editing `package.json`) is per-project, obviously.

> If one has done a global npm install of yarn, would the recommendation be to npm uninstall --global yarn and continue to use it via Corepack?

Yes.

Having multiple `yarn` binaries in `PATH` have caused problems for some corepack users. I think it is worth adding a clarification to the installation guide.
